services/src/datasets/external/gbif.rs

```diff
@@ -256,6 +570,103 @@ impl GbifDataProvider {
         Ok(items)
     }
 
+    async fn get_filter_search_items(
+        &self,
+        limit: &u32,
+        offset: &u32,
+        path: &str,
+        search_string: &str,
+    ) -> Result<Vec<CollectionItem>> {
+        let items = self
+            .query_filter_items(limit, offset, path, search_string)
+            .await?
+            .into_iter()
+            .map(|name| {
+                let new_path = GbifDataProvider::extend_path((*path).to_string(), &name);
+                let new_id = "select/".to_string() + &new_path;
+
+                CollectionItem::Collection(LayerCollectionListing {
+                    id: ProviderLayerCollectionId {
+                        provider_id: GBIF_PROVIDER_ID,
+                        collection_id: LayerCollectionId(new_id),
+                    },
+                    name,
+                    description: String::new(),
+                    properties: Default::default(),
+                })
+            })
+            .collect::<Vec<_>>();
+
+        Ok(items)
+    }
+
+    async fn get_filter_autocomplete_items(
+        &self,
+        limit: &u32,
+        offset: &u32,
+        path: &str,
+        search_string: &str,
+    ) -> Result<Vec<String>> {
+        let filter_items = timeout(
+            Duration::from_secs(self.autocomplete_timeout),
+            self.query_filter_items(limit, offset, path, search_string),
+        )
+        .await;
+
+        match filter_items {
+            Ok(filter_items) => Ok(filter_items?),
+            Err(_) => Ok(Vec::<String>::new()),
+        }
+    }
+
+    async fn query_filter_items(
+        &self,
+        limit: &u32,
+        offset: &u32,
+        path: &str,
+        search_string: &str,
+    ) -> Result<Vec<String>> {
+        let column = GbifDataProvider::level_name(path);
+        if !Self::LEVELS.contains(&column.as_str()) {
+            return Err(Error::InvalidLayerCollectionId);
+        }
+        let filters = GbifDataProvider::get_filters(path);
+        let conn = self.pool.get().await?;
+        let query = &format!(
+            r#"
+            SELECT DISTINCT "{column}"
```

Comment:
why is the distinct needed here?

Reply:
There can be duplicates otherwise. Since we're selecting one item as filter duplicate entries are not wanted